File: py/cake.py
import heapq
from functools import reduce

candles = [3, 2, 1, 5,5,5, 3] 

#using lambda and list comprehension
get_highest_num = lambda arr: len([e for e in arr if e == max(arr)])


#using functional reduce
vget_highest_num = lambda arr: reduce(max, arr)

#imperative version
def xget_highest_num(arr):
    ar = []
    for e in arr:
        if e is max(arr):
            ar.append(e)
    return len(ar)


print(vget_highest_num(candles))

File: js/cake.js

let candles = [3, 2, 1, 10, 10,]
let length = 4

//functional version
const countHighestNumbers = arr => 
        arr.filter((number)=> (number === arr.reduce(
            (a,b)=> Math.max(a,b)))).length

//imperative version
function xcountHighestNumbers(arr){
    let highestHeight = []
    for(var i in arr){
        if(arr[i] === arr.reduce(function(a,b){
            return Math.max(a,b)
        }))
        {
            highestHeight.push(arr[i])
        }
    }
}


console.log(countHighestNumbers(candles))